packages/bundler/src/test/es6-module-bundling-scenarios_test.ts

```diff
@@ -199,9 +199,7 @@ suite('Es6 Module Bundling', () => {
           await bundler.bundle(await bundler.generateManifest([aUrl]));
       assert.deepEqual(documents.get(aUrl)!.content, heredoc`
         async function go() {
-          const b = await import('./b.js').then(({
-            $b
-          }) => $b);
+          const b = await import('./b.js').then(bundle => bundle && bundle.$b);
```

Comment:
Should this be: `bundle => bundle && bundle.$b || {}`?

Because it's totally legal to do:

```javascript
const b = await import('./b.js');
b.thisPropertyDoesNotExist === undefined
```

Reply:
I mean ideally AMD would make the default export `{}` so that `$b` would be {} so long as `./b.js` exists, regardless of whether it exports anything

Reply:
So without bundling, `import('./foo.js')` will resolve to an empty `Module` object (which you could attempt to get an undefined property from), but the AMD transform is a Promise that resolves to `undefined` (which you can't destructure). By giving you `undefined` when importing bundles, this is consistent with the AMD transform.

Another possible fix would be to make sure AMD transform always gives you an empty object, but that would need to be done in [the Babel transform](https://github.com/babel/babel/blob/a4795408b47e713aa2635d54197343be77e30f35/packages/babel-plugin-transform-modules-amd/src/index.js#L46) AFAICT

Reply:
To be clear, that wouldn't work in the unbundled, AMD transform case either (e.g. at dev time with `polymer serve` on a non-ES modules browser like Firefox) - if a developer forgets to export something, the destructure call will still error. That is, no guarantees are provided by us if you try to use something that isn't exported.

Reply:
@keanulee after reviewing, the bundle-producing code is probably not worth changing as it seems correct.  I don't exactly mind having a safety in here like you've done to guard against errors introduced by bundler's dynamic import statement transform not playing nice with the AMD transform, but this ultimately feels like a problem with that transform.

Also there's a subtle issue with respect to what you said...
>  if a developer forgets to export something, the destructure call will still error

...because it isn't actually an error to destructure a property from a module object if that name  doesn't exist.  It's an error to try and import a non-exported name:
```
import {notAThing} from 'something'; // error
```
...but it is okay to destructure a value from a module object that is not exported, which is functionally what a dynamic import case is like:
```
import * as stuff from 'something';
const {notAThing} = stuff; // not an error
```
However, if we make a specific case for `|| {}` in the dynamic import `.then` code we are maybe being inconsistent, since a non-dynamic import of `$b` as in the following would be (at best?) an undefined value after AMD transform instead of an empty object:
```
// originally: import * as stuff from './b.js';
import {$b as stuff} from './bundle.js';
stuff.value // error in AMD transform output when value not exported
```

So with all that reasoning... I will approve this PR since it addresses the most common problem which is dynamically imported fragment that doesn't export anything blowing up after bundling.  The next step is addressing the behavior of the AMD transform so it doesn't cause downstream problems for transforming modules without exports.